pyrogram/sync.py

```diff
@@ -40,23 +40,33 @@ def async_to_sync_wrap(*args, **kwargs):
         try:
             loop = asyncio.get_event_loop()
         except RuntimeError:
-            loop = main_loop
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
```

Comment:
Adding `asyncio.set_event_loop(loop)` here fixed an issue where a new loop would be created every time a method was called in the same new thread.

Reply:
yes. its a good point :)